experimental/dds/tree2/src/feature-libraries/editable-tree/utilities.ts

```diff
@@ -99,3 +99,23 @@ export function getStableNodeKey(
 		return brand(id);
 	}
 }
+
+/**
+ * Checks whether or not a given path is parented under the root field.
+ * @param path - the path you want to check.
+ * @returns a boolean on whether or not the provided path is parented under the root field.
+ */
+export function isParentedUnderRootField(path: UpPath): boolean {
+	let currentPath = path;
+	while (currentPath.parentField !== undefined) {
+		if (currentPath.parentField === rootFieldKey) {
```

Comment:
This isn't the correct way to do this check. A path that has this key somewhere along the path (but not as the root) should not pass this check.

I'd suggest adding a regression test for this case, but if you do the suggested change to getDetachedSequenceContainingPath I mentioned above, its impossible to have (or test for) this bug.

Reply:
Refactored to `getDetachedFieldContainingPath` as suggested above